### PR TITLE
Patch GRUB config: Add --no-floppy option

### PR DESCRIPTION
**Pull request by johnliu55tw**
related issue: https://github.com/harvester/harvester/issues/2115

And this is just a temp workaround for https://github.com/rancher-sandbox/cOS-toolkit/pull/1220

### CODE REVIEW

**Comment by tlehman** [comment]
All this does is [prevent GRUB from searching floppy drives](https://www.gnu.org/software/grub/manual/grub/grub.html#:~:text=The%20%2D%2Dno%2Dfloppy%20option%20prevents%20searching%20floppy%20devices%2C%20which%20can%20be%20slow.). 

Looks good to me

**Review by weihanglo — APPROVED** [review]
Code looks fine, and double checked existing grub.cfg. The sed part seems solid enough. Tho I am not sure how to conduct an e2e test for it.

File: package/harvester-os/files/usr/sbin/harv-install [hunk 1]
```diff
@@ -260,7 +260,7 @@ add_debug_grub_entry()
 {
     cat > "${STATEDIR}/grubmenu" << "EOF"
 menuentry "${display_name} (debug)" --id debug {
-  search.fs_label COS_STATE root
+  search --no-floppy --label COS_STATE root
```
Comment (weihanglo) [inline package/harvester-os/files/usr/sbin/harv-install:263]:
```suggestion
  search --no-floppy --label --set=root COS_STATE
```

File: package/harvester-os/files/usr/sbin/harv-install [hunk 2]
```diff
@@ -290,6 +290,13 @@ update_grub_settings()
         sed -i "s/^set crash_kernel_params=.*/crash_kernel_params=\"${CRASH_KERNEL_PARAMS}\"/" ${TARGET}/etc/cos/bootargs.cfg
     fi
 
+    # PATCH: Adding '--no-floppy' option to search command
+    GRUB_CFG="${STATEDIR}/grub2/grub.cfg"
+    sed -i "s/search /search --no-floppy /" "$GRUB_CFG"
+    sed -i "s/search.fs_label /search --no-floppy --label /" "$GRUB_CFG"
+    sed -i "s/search.file /search --no-floppy --file /" "$GRUB_CFG"
+    sed -i "s/search.fs_uuid /search --no-floppy --fs-uuid /" "$GRUB_CFG"
```
Comment (weihanglo) [inline package/harvester-os/files/usr/sbin/harv-install:298]:
```suggestion
    sed -i "s/search /search --no-floppy --set=root /" "$GRUB_CFG"
    sed -i "s/search.fs_label /search --no-floppy --set=root --label /" "$GRUB_CFG"
    sed -i "s/search.file /search --no-floppy --set=root --file /" "$GRUB_CFG"
    sed -i "s/search.fs_uuid /search --no-floppy --set=root --fs-uuid /" "$GRUB_CFG"
```

**Review by bk201 — APPROVED** [review]

**Comment by bk201** [comment]
@johnliu55tw Do we still need this after catching up on cOS-toolkit?

**Comment by johnliu55tw** [comment]
cOS-toolkit has fixed this issue, and seems like this fix has nothing to do with https://github.com/harvester/harvester/issues/2115. I'd say we can revert this.

**Pull request merged**